numpy/lib/npyio.py

```diff
@@ -1106,18 +1106,20 @@ def savetxt(fname, X, fmt='%.18e', delimiter=' ', newline='\n', header='',
 
         # Handle 1-dimensional arrays
         if X.ndim == 1:
-            # Common case -- 1d array of numbers
-            if X.dtype.names is None:
-                X = np.atleast_2d(X).T
-                ncol = 1
+            X = np.atleast_2d(X).T
 
-            # Complex dtype -- each field indicates a separate column
-            else:
-                ncol = len(X.dtype.descr)
-        else:
-            ncol = X.shape[1]
+        nrow = X.shape[0]
```

Comment:
I don't think this covers all the cases
```
In [28]: dt = dtype([('a','i1'),('b', 'i2'),('','i4')], align=True)

In [29]: a = array([(1,2,3),(4,5,6)], dt)

In [30]: a.shape
Out[30]: (2,)

In [31]: len(a.dtype.descr)
Out[31]: 4

In [32]: b = atleast_2d(a)

In [33]: b.shape
Out[33]: (1, 2)

In [34]: len(b.dtype.descr)
Out[34]: 4
```
The `4` is because an alignment column has been inserted. That may also be a problem, as there are only 3 data columns.